tests/test_widgets.py

```diff
@@ -2184,6 +2183,72 @@ def testNoneSamplePropertiesDialog(self):
         self.assertEqual('0', self.dialog.sample_property_table.item(0, 1).text())
 
 
+class TestInsertPrimitiveDialog(unittest.TestCase):
```

Comment:
This only tests the default primitive arguments, you should also add tests where the values are changed and a test for validation if value is zero or any other invalid. (For these extra tests, you can test 2 primitive instead of all of them)

Reply:
Added additional tests